lib/readline.js

```diff
@@ -456,6 +457,23 @@ Interface.prototype._insertString = function(c) {
 
     // a hack to get the line refreshed if it's needed
     this._moveCursor(0);
+    // emit current line for generating preview
+    this.emit('buffer', this.line);
+  }
+};
+
+Interface.prototype.appendPreview = function(result) {
+  this.previewResult = ` // ${result}`;
+  this.output.write(this.previewResult);
+  // move the cursor to the line end
+  cursorTo(this.output, this._getCursorPos().cols);
+};
+
+// Refresh the line if preview present
+Interface.prototype.clearPreview = function() {
```

Comment:
These should either be hidden (think: `Symbol` access), or documented. I'm not sure these make sense in readline, so maybe they should be in repl or private?

Reply:
I guess making it hidden with `Symbol` access would do good.